Formula/notcurses.rb

```diff
@@ -0,0 +1,37 @@
+class Notcurses < Formula
+  desc "Blingful character graphics/TUI library"
+  homepage "https://nick-black.com/dankwiki/index.php/Notcurses"
+  url "https://github.com/dankamongmen/notcurses/archive/refs/tags/v2.3.12.tar.gz"
+  sha256 "ce042908fac11f7df1f9eaa610e46e9c615f53ab036b7c27ae2396292512407b"
+  license "Apache-2.0"
+
+  depends_on "cmake" => :build
+  depends_on "doctest" => :build
+  depends_on "pandoc" => :build
+  depends_on "pkg-config" => :build
+  depends_on "ffmpeg"
+  depends_on "libunistring"
+  depends_on "ncurses"
+  depends_on "readline"
+  uses_from_macos "zlib"
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args + %w[
+      -DBUILD_TESTING=on
```

Comment:
Can we do without this?

Reply:
probably? everything up through the `%w` was output from `brew create --cmake`, so we might want to pull it out of that template as well. the `-DBUILD_TESTING=on` is necessary to build `notcurses-tester`, as the Homebrew CMake process explicitly exports `-DBUILD_TESTING=off`, for reasons i can't quite fathom.

the `-B` can go away, but otherwise we need a `mkdir + cd` in its place, or to build in the toplevel. i can do the latter if you'd prefer.

the `-S` is, i think, strictly unnecessary.

Reply:
hrmmm, yeah `brew install --build-from-source` was unhappy with either `-B build` removed, `-S .` removed, or `-S .` replaced with `.`. =[

Reply:
Oh, I know what's in the formula template. I was asking about `-DBUILD_TESTING=on`.

Reply:
sorry, didn't mean to come across obnoxious. `notcurses-tester` runs the complete unit testing suite:

```
```

`notcurses-info` is a tool endusers would run; i can add that. but it's a library; the end-user is developers. there are two programs included, `notcurses-input` and `nctetris`, which would be end-user programs, but they wouldn't be particularly easy/informative to wrap as e2e tests, i think...

Reply:
also there are a bunch of PoCs (proofs of concept) that are run to inspect/demonstrate some small things, but not installed; I supply `-DUSE_POC=off` to disable building them in packages. but if you want, i could strike that, and several of them are run as part of our default `ctest`; i could explicitly add them here.

Reply:
Not particularly easy/informative to wrap as e2e tests is fine -- we don't want e2e tests. We just want a simple sanity check to see that the software being installed isn't completely broken. This is why we pass `-DBUILD_TESTING=OFF` in `std_cmake_args`.Most users aren't interested in unit tests, and we don't really have the CI resources to run them for all formulae we build.

Reply:
thanks a lot for the explanation. i'm going to go ahead and switch to `notcurses-info` then (justified in the comment below).

Reply:
i've moved over to `notcurses-info`, and eliminated the extra CMake arguments.